File: 1776-minimum-operations-to-reduce-x-to-zero/minimum-operations-to-reduce-x-to-zero.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, int x) {
        // Thinking in reverse
        int total=0;
        for(int num:nums) total+=num;
        
        int i=0,sum=0,remaining=total-x;
        int len=-1;
        for(int j=0;j<nums.size();++j){
            sum+=nums[j];
            while(sum>remaining && i<=j) sum-=nums[i++];
            if(sum==remaining) len=max(len,j-i+1);
        }
        return (len==-1)?len:nums.size()-len;
    }
};

File: 1776-minimum-operations-to-reduce-x-to-zero/minimum-operations-to-reduce-x-to-zero.py
class Solution:
    def minOperations(self, nums: list[int], x: int) -> int:
        # When we reverse our thinking we can look for longest subarray with sum sum(nums)-x

        n=len(nums)
        total = sum(nums)
        target = total-x

        summ=0
        sol=float('inf')
        l=0

        for r in range(n):
            summ+=nums[r]

            while summ>target and l<=r:
                summ-=nums[l]
                l+=1
            
            if summ==target:
                length = (l-0)+(n-1-r)
                sol = min(sol,length)
        
        return -1 if sol==float('inf') else sol